test/components/views/elements/TagComposer-test.tsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import {
+    act,
+    renderIntoDocument,
+    findRenderedDOMComponentWithTag,
+    Simulate,
+    scryRenderedDOMComponentsWithClass,
+} from 'react-dom/test-utils';
+
+import TagComposer from "../../../../src/components/views/elements/TagComposer";
+
+describe("<TagComposer />", () => {
+    const defaultProps = {
+        tags: [],
+        onAdd: jest.fn(),
+        onRemove: jest.fn(),
+    };
+    const getComponent = (props = {}) => renderIntoDocument<React.Component>(
+        <TagComposer {...defaultProps} {...props} />,
+    ) as React.Component;
+
+    const getForm = container => findRenderedDOMComponentWithTag(container, 'form');
+    const getInput = container => {
+        const form = getForm(container);
+        return form.querySelector('input');
+    };
+
+    const getAddButton = container => {
+        const form = getForm(container);
+        return form.querySelector('[role=button]');
+    };
+
+    const editNewTag = async (container, value: string) => act(async () => {
+        const input = getInput(container);
+        Simulate.change(input, { target: { value } as unknown as EventTarget });
+    });
+
+    const submitForm = async (container) => act(async () => {
+        const form = getForm(container);
+        Simulate.submit(form);
+    });
+
+    // Field component debounces validate on change
+    // jest@26 fake timers and lodash debounce are not friends
+    // blur input to trigger a validation
+    const blurInput = async (container) => act(async () => {
+        const input = getInput(container);
+        Simulate.blur(input);
+    });
+
+    const clickAddNewTag = async (container) => act(async () => {
+        Simulate.click(getAddButton(container));
+    });
+
+    it('renders tags', () => {
+        const tags = ['tag 1', 'tag 2'];
+        const container = getComponent({ tags });
+
+        const tagElements = scryRenderedDOMComponentsWithClass(container, 'mx_Tag');
+        expect(tagElements).toMatchSnapshot();
```

Comment:
can we inspect the returned elements instead of using a snapshot? Snapshots are unfortunately harder to work with.